sdk/synapse/synapse-artifacts/review/synapse-artifacts.api.md

```diff
@@ -0,0 +1,5227 @@
+## API Report File for "@azure/synapse-artifacts"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import * as coreHttp from '@azure/core-http';
+import { HttpMethods } from '@azure/core-http';
+import { HttpOperationResponse } from '@azure/core-http';
+import { OperationArguments } from '@azure/core-http';
+import { OperationOptions } from '@azure/core-http';
+import { OperationSpec } from '@azure/core-http';
+import { PagedAsyncIterableIterator } from '@azure/core-paging';
+import { PipelineOptions } from '@azure/core-http';
+import { Poller } from '@azure/core-lro';
+import { PollOperationState } from '@azure/core-lro';
+import { RestResponse } from '@azure/core-http';
+import { TokenCredential } from '@azure/core-http';
+
+// @public
+export interface Activity {
+    [property: string]: any;
+    dependsOn?: ActivityDependency[];
+    description?: string;
+    name: string;
+    type: "Container" | "Execution" | "Copy" | "HDInsightHive" | "HDInsightPig" | "HDInsightMapReduce" | "HDInsightStreaming" | "HDInsightSpark" | "ExecuteSSISPackage" | "Custom" | "SqlServerStoredProcedure" | "ExecutePipeline" | "Delete" | "AzureDataExplorerCommand" | "Lookup" | "WebActivity" | "GetMetadata" | "IfCondition" | "Switch" | "ForEach" | "AzureMLBatchExecution" | "AzureMLUpdateResource" | "AzureMLExecutePipeline" | "DataLakeAnalyticsU-SQL" | "Wait" | "Until" | "Validation" | "Filter" | "DatabricksNotebook" | "DatabricksSparkJar" | "DatabricksSparkPython" | "SetVariable" | "AppendVariable" | "AzureFunctionActivity" | "WebHook" | "ExecuteDataFlow" | "SynapseNotebook" | "SparkJob" | "SqlPoolStoredProcedure";
+    userProperties?: UserProperty[];
+}
+
+// @public
+export interface ActivityDependency {
+    [property: string]: any;
```

Comment:
okay I notice this is a common problem in many types. Is this a codegen/swagger issue?

Reply:
If I had to guess the schema for these types has `additionalProperties: true` along with defined properties? Unless there is very strong rationale for this it should be avoided because we're effectively not getting any type checking at all on these objects.